e2e/dockerv2/main.go

```diff
@@ -1,34 +1,38 @@
 package main
 
 import (
+	"errors"
 	"fmt"
 	"log"
 	"net/http"
+	"os/exec"
 )
 
 type data struct {
-	host string
-	path string
+	host  string
+	path  string
+	image string
 }
 
+type Kind string
+
+const (
```

Comment:
Why the abstraction of Kind?

Reply:
Just thought it may look better than direct string comparison. Will change it if it's unnecessary.

Reply:
It makes the overall code a bit harder to understand in my view as the string for each type wouldn't change frequently anyway. Ok to keep it as is, but feel like this might be an over optimization for nicer code > readability. What do you think?

Reply:
Well, It would've been useful if we had lots of types and not only pull and push. And since they're not gonna get changed at all I guess going with strings instead of Kind type is better rn.